utils/createUrl.ts

```diff
@@ -1,67 +1,23 @@
-/**
- * Erstellt eine Such-URL für die offizielle DB-Website
- * @param {Object} params - Suchparameter
- * @param {string} params.from - Name der Startstation
- * @param {string} params.to - Name der Zielstation
- * @param {string} params.date - Abfahrtsdatum im ISO-Format
- * @param {number} params.class - Reiseklasse (1 oder 2)
- * @param {Object} params.fromStation - Startstation-Objekt mit ID und Koordinaten
- * @param {Object} params.toStation - Zielstation-Objekt mit ID und Koordinaten
- * @returns {string} DB-Website Such-URL
- */
-export function createDBSearchUrl({
-	from,
-	to,
-	date,
-	class: travelClass = 2,
-	fromStation = null,
-	toStation = null,
-}) {
-	// Datum formatieren für DB-URL
-	const formattedDate = formatDate(date);
-	const parts = [
-		"sts=true",
-		`so=${encodeURIComponent(from)}`,
-		`zo=${encodeURIComponent(to)}`,
-		`kl=${travelClass}`,
-		"r=13:16:KLASSENLOS:1",
-	];
+import type { VendoJourney } from "./schemas";
 
-	// Hilfsfunktion zum Hinzufügen von Stationsdaten
-	const addStation = (prefix, station) => {
-		if (!station?.id) return;
-		parts.push(`${prefix}oid=${createStationId(station)}`);
-		parts.push(`${prefix}ei=${station.id}`);
-	};
-
-	// Start- und Zielstation hinzufügen
-	addStation("so", fromStation);
-	addStation("zo", toStation);
-	parts.push("sot=ST", "zot=ST");
-
-	// Weitere Parameter für die DB-Suche hinzufügen
-	parts.push(
-		`hd=${formattedDate}`,
-		"hza=D",
-		"hz=%5B%5D",
-		"ar=false",
-		"s=false",
-		"d=false",
-		"vm=00,01,02,03,04,05,06,07,08,09",
-		"fm=false",
-		"bp=false",
-		"dlt=false",
-		"dltv=false"
-	);
-	return `https://www.bahn.de/buchung/fahrplan/suche#${parts.join("&")}`;
+interface Station {
+	id?: string;
+	stationId?: string;
+	uicCode?: string;
+	evaId?: string;
+	name: string;
+	longitude?: number;
+	latitude?: number;
+	x?: number;
+	y?: number;
```

Comment:
Why are all of these properties except `name` optional?

Reply:
Because the code accessing these properties handles them as such, e.g. line 154 same file. (Logic is mostly unchanged from current main branch)